packages/cms/lists/views/footer-social-media-links.tsx

```diff
@@ -0,0 +1,332 @@
+  icon: '',
+  link: '',
+}
+
+const SocialMediaLinkComponent = (props: {
+  socialMediaLink: SocialMediaLink
+  onIconChange: (value: string) => void
+  onLinkChange: (e: ChangeEvent<HTMLInputElement>) => void
+  actionElement: ReactNode
+  isDisabled?: boolean
+  showDragHandle?: boolean
+}) => {
+  const { socialMediaLink, isDisabled = false, showDragHandle = false } = props
+  return (
+    <ItemContainer>
+      {showDragHandle && <DragHandle />}
+      <IconSelect>
+        <Select
+          placeholder="請選擇 icon（必填）"
+          value={
+            socialMediaLink.icon
+              ? SOCIAL_MEDIA_OPTIONS.find(
+                  (opt) => opt.value === socialMediaLink.icon
+                ) || null
+              : null
+          }
+          onChange={(option) => {
+            if (option) {
+              props.onIconChange(option.value)
+            }
+          }}
+          options={SOCIAL_MEDIA_OPTIONS}
+          isDisabled={isDisabled}
+        />
+      </IconSelect>
+      <LinkInput>
+        <TextInput
+          placeholder="連結網址（必填）"
+          onChange={props.onLinkChange}
+          value={socialMediaLink.link}
+          disabled={isDisabled}
+        />
+      </LinkInput>
+      {props.actionElement}
+    </ItemContainer>
+  )
+}
+
+export const Field = ({
+  field,
+  value,
+  onChange,
+}: FieldProps<typeof controller>) => {
+  const parseSocialMediaLinks = (
+    val: string | null | undefined
+  ): SocialMediaLink[] => {
+    if (!val) return []
+    try {
+      const parsed = JSON.parse(val)
+      return Array.isArray(parsed) ? parsed : []
+    } catch {
+      return []
+    }
+  }
+
+  const [socialMediaLinks, setSocialMediaLinks] = useState<SocialMediaLink[]>(
+    parseSocialMediaLinks(value)
+  )
+  const [newLink, setNewLink] = useState<SocialMediaLink>({
+    ...socialMediaLinkTemplate,
+  })
+  const [prevValue, setPrevValue] = useState(value)
+
+  const sensors = useSensors(
+    useSensor(PointerSensor, {
+      activationConstraint: {
+        distance: 8, // Minimum distance required before dragging starts
+      },
+    }),
+    useSensor(KeyboardSensor, {
+      coordinateGetter: sortableKeyboardCoordinates,
+    })
+  )
+
+  if (value !== prevValue) {
+    setPrevValue(value)
+    setSocialMediaLinks(parseSocialMediaLinks(value))
+  }
+
+  const onAddNewLink = () => {
+    if (onChange && newLink.link.trim()) {
+      const newLinks = [...socialMediaLinks, newLink]
+      setSocialMediaLinks(newLinks)
+      onChange(JSON.stringify(newLinks))
+      setNewLink({ ...socialMediaLinkTemplate })
+    }
+  }
+
+  const onDeleteLink = (index: number) => {
+    if (onChange && index >= 0 && index < socialMediaLinks.length) {
+      const newLinks = [...socialMediaLinks]
+      newLinks.splice(index, 1)
+      setSocialMediaLinks(newLinks)
+      onChange(JSON.stringify(newLinks))
+    }
+  }
+
+  const onUpdateLinkIcon = (index: number, icon: string) => {
```

Comment:
would we need to check if icon has value if it's marked as required with placeholder on L129 ?